spark/src/main/scala/org/apache/spark/sql/delta/commands/CreateDeltaTableCommand.scala

```diff
@@ -429,7 +430,7 @@ case class CreateDeltaTableCommand(
         // existing metadata because new table desc will not have related metadata assigned yet
         val differences = SchemaUtils.reportDifferences(
           dropColumnMappingMetadata(existingMetadata.schema),
-          tableDesc.schema)
+          CharVarcharUtils.replaceCharVarcharWithStringInSchema(tableDesc.schema))
```

Comment:
@ldadima I see that `SchemaUtils.reportDifferences` already converts the `char` to `varchar` in the schema given as the second argument. [Here](https://github.com/delta-io/delta/blob/master/spark/src/main/scala/org/apache/spark/sql/delta/schema/SchemaUtils.scala#L476) is the code pointer.

Do we need to do the same here? I also tried running your test and I see it passes with this code change here.

Reply:
@vkorukanti Yes. You are right, it's my mistake. This test works for master branch, but doesn't work for 2.3.0 version. This changes don't need for master branch.